server/handlers/handlers.go

```diff
@@ -80,26 +82,39 @@ func New(plugin PluginIface) *ActivityHandler {
 }
 
 func (ah *ActivityHandler) Start() {
+	ah.quitting.Store(false)
+
 	// This is constant for now, but report it as a metric to future proof dashboards.
 	ah.plugin.GetMetrics().ObserveChangeEventQueueCapacity(activityQueueSize)
 
-	for i := 0; i < numberOfWorkers; i++ {
-		go func() {
-			for {
-				select {
-				case activity := <-ah.queue:
-					ah.plugin.GetMetrics().DecrementChangeEventQueueLength(activity.ChangeType)
-					ah.handleActivity(activity)
-				case <-ah.quit:
-					// we have received a signal to stop
-					return
-				}
+	// doStart is the meat of the activity handler worker
+	doStart := func() {
+		for {
+			select {
+			case activity := <-ah.queue:
+				ah.plugin.GetMetrics().DecrementChangeEventQueueLength(activity.ChangeType)
+				ah.handleActivity(activity)
+			case <-ah.quit:
+				// we have received a signal to stop
+				return
 			}
-		}()
+		}
+	}
+
+	// isQuitting informs the recovery handler if the shutdown is intentional
+	isQuitting := func() bool {
+		return ah.quitting.Load()
+	}
+
+	logError := ah.plugin.GetAPI().LogError
```

Comment:
A bit coarse, as I'd prefer to just send a logging interface, but keeping it simple until we can tackle some of the existing ergonomics.